### PR TITLE
config: houseOwnedByAccount

### LINKED ISSUE #2879: [Suggestion] New house settings
It seems that currently we have no limitations to buying houses other than one per character, and since recently tibia added the possibility to own 3 houses on the same character (or account/world), here is my list of suggestions regarding our current house configs:

- [ ] House per account limit instead of per character
- [x] Add option to show house price on look at door = true/false
- [x] Level requirement
- [x] Premium requirement
- [x] a setting to make all characters on account invited on the house by default = true/false

### PR DESCRIPTION
**Pull request by MillhioreBT**
This enables or disables the possibility that characters from the same account of the home owner can have access.

Closes #2879

### CODE REVIEW

File: config.lua.dist [hunk 1]
```diff
@@ -42,6 +42,7 @@ deathLosePercent = -1
 -- NOTE: set housePriceEachSQM to -1 to disable the ingame buy house functionality
 housePriceEachSQM = 1000
 houseRentPeriod = "never"
+houseAccountOwner = false
```
Comment (DSpeichert) [inline config.lua.dist:45]: Maybe `houseOwnedByAccount` would convey the meaning better?

File: src/iologindata.cpp [hunk 2]
```diff
@@ -160,6 +160,19 @@ uint32_t IOLoginData::gameworldAuthentication(const std::string& accountName, co
 	return accountId;
 }
 
+uint32_t IOLoginData::getAccountIdByPlayerName(const std::string& playerName)
```
Comment (Lyuzera) [inline src/iologindata.cpp:163]: `bool House::isInvited(const Player* player)` is checked frequently, maybe needs a cache on Game class to improve performance.

File: config.lua.dist (hunk 1 above)
Comment (MillhioreBT) [reply config.lua.dist:45]: > Maybe `houseOwnedByAccount` would convey the meaning better?
Yes, totally better!

File: src/iologindata.cpp (hunk 2 above)
Comment (MillhioreBT) [reply src/iologindata.cpp:163]: > `bool House::isInvited(const Player* player)` is checked frequently, maybe needs a cache on Game class to improve performance.

I like the idea, I'll try it thanks for the review

Comment (Lyuzera) [reply src/iologindata.cpp:163]: i thought add ```uint32_t ownerAccount = 0;``` in House class, i tested and looks pretty good:

```c
// AccessHouseLevel_t House::getHouseAccessLevel(const Player* player)
	if (ownerAccount == player->getAccount()) {
		return HOUSE_OWNER;
	}
```


```diff
@@ void House::setOwner(uint32_t guid, bool updateDatabase/* = true*/, Player* player/* = nullptr*/)
{
        ....

		//clean access lists
		owner = 0;
+		ownerAccount = 0;
		setAccessList(SUBOWNER_LIST, "");
		setAccessList(GUEST_LIST, "");

    ....

	if (guid != 0) {
		std::string name = IOLoginData::getNameByGuid(guid);
		if (!name.empty()) {
			owner = guid;
			ownerName = name;
+			ownerAccount = IOLoginData::getAccountIdByPlayerName(name);
		}
	}
```

Comment (MillhioreBT) [reply src/iologindata.cpp:163]: It definitely looks great much better.

File: config.lua.dist (hunk 1 above)
Comment (DSpeichert) [reply config.lua.dist:45]: I think you forgot to update this file.

File: src/configmanager.cpp [hunk 3]
```diff
@@ -150,6 +150,7 @@ bool ConfigManager::load()
 	boolean[ONLINE_OFFLINE_CHARLIST] = getGlobalBoolean(L, "showOnlineStatusInCharlist", false);
 	boolean[YELL_ALLOW_PREMIUM] = getGlobalBoolean(L, "yellAlwaysAllowPremium", false);
 	boolean[FORCE_MONSTERTYPE_LOAD] = getGlobalBoolean(L, "forceMonsterTypesOnLoad", true);
+	boolean[HOUSE_OWNED_BY_ACCOUNT] = getGlobalBoolean(L, "houseOwnedByAccount ", false);
```
Comment (DSpeichert) [inline src/configmanager.cpp:153]: space in key string?

File: src/house.h [hunk 4]
```diff
@@ -248,6 +248,7 @@ class House
 
 		uint32_t id;
 		uint32_t owner = 0;
+		uint32_t ownerAccount = 0;
```
Comment (DSpeichert) [inline src/house.h:251]: `ownerAccountId`?

File: src/house.cpp [hunk 5]
```diff
@@ -79,6 +79,7 @@ void House::setOwner(uint32_t guid, bool updateDatabase/* = true*/, Player* play
 
 		//clean access lists
 		owner = 0;
+		ownerAccount = 0;
```
Comment (DSpeichert) [inline src/house.cpp:82]: `ownerAccountId`?

**Comment by omarcopires** [comment]
I've been using it on my production server for a few days, and I haven't had any problems. The code works perfectly, good work by the author.

**Comment by Lyuzera** [comment]
> I've been using it on my production server for a few days, and I haven't had any problems. The code works perfectly, good work by the author.

me too, using on my server with 120 players online and at the moment is working fine.

**Comment by DSpeichert** [comment]
This won't be merged until the requested changes are made.

**Comment by omarcopires** [comment]
> This won't be merged until the requested changes are made.

He has already made corrections, just forgot to mark them as resolved.

File: src/configmanager.cpp [hunk 6]
```diff
@@ -150,6 +150,7 @@ bool ConfigManager::load()
 	boolean[ONLINE_OFFLINE_CHARLIST] = getGlobalBoolean(L, "showOnlineStatusInCharlist", false);
 	boolean[YELL_ALLOW_PREMIUM] = getGlobalBoolean(L, "yellAlwaysAllowPremium", false);
 	boolean[FORCE_MONSTERTYPE_LOAD] = getGlobalBoolean(L, "forceMonsterTypesOnLoad", true);
+	boolean[HOUSE_OWNED_BY_ACCOUNT] = getGlobalBoolean(L, "houseOwnedByAccount", false);
```
Comment (DSpeichert) [inline src/configmanager.cpp:153]: The config.lua.dist still shows `houseAccountOwner`

**Pull request merged**